src/mesh/Draco_Mesh.hh

```diff
@@ -0,0 +1,126 @@
+ * mesh with layout (cell adjacency) information.  This class also provides
+ * basic services, including access to cell information.  This mesh is based on
+ * an unstructured mesh implementation by Kent Budge.
+ *
+ * Two important features for a fully realized Draco_Mesh are the following:
+ * 1) Layout, which stores cell connectivity and hence the mesh topology.
+ *    a) It has an internal layout containing local cell-to-cell linkage,
+ *    b) and a boundary layout with side and off-process linkage.
+ * 2) Geometry, which implies a metric for distance between points.
+ *
+ * Possibly temporary features:
+ * 1) The cell_type_ vector (argument to the constructor) is currently taken to
+ *    be the number of nodes per cell.
+ * 2) The layout data structure(s) will proabably be moved to a separate class,
+ *    where accessors might be used on a flattened version.
+ */
+//===========================================================================//
+
+class Draco_Mesh {
+public:
+  // >>> TYPEDEFS
+  typedef rtt_mesh_element::Geometry Geometry;
+  typedef std::map<unsigned,
+                   std::vector<std::pair<unsigned, std::vector<unsigned>>>>
+      Layout;
+  typedef std::map<unsigned, std::vector<std::pair<int, std::vector<unsigned>>>>
+      Boundary_Layout;
+
+private:
+  // >>> DATA
+
+  // Dimension
+  const unsigned dimension;
+
+  // Geometry enumeration
+  const Geometry geometry;
+
+  // Number of cells
+  const unsigned num_cells;
+
+  // Number of nodes
+  const unsigned num_nodes;
+
+  // Side set flag (can be used for mapping BCs to sides)
+  const std::vector<unsigned> side_set_flag;
+
+  // Layout of mesh: vector index is cell index, vector element is
+  // description of cell's adjacency to other cells in the mesh.
+  Layout cell_to_cell_linkage;
+  // TODO: update this to a full layout class.
+
+  // Boundary layout of mesh
+  Boundary_Layout cell_to_side_linkage;
+
+  // vector subscripted with node index with coordinate vector
+  std::vector<std::vector<double>> node_coord_vec;
+
+public:
+  //! Constructor.
+  Draco_Mesh(unsigned dimension_, Geometry geometry_,
+             const std::vector<unsigned> &cell_type_,
+             const std::vector<unsigned> &cell_to_node_linkage_,
+             const std::vector<unsigned> &side_set_flag_,
+             const std::vector<unsigned> &side_node_count_,
+             const std::vector<unsigned> &side_to_node_linkage_,
+             const std::vector<double> &coordinates_,
+             const std::vector<unsigned> &global_node_number_);
+
+  // >>> ACCESSORS
+
+  unsigned get_dimension() const { return dimension; }
+  Geometry get_geometry() const { return geometry; }
+  unsigned get_num_cells() const { return num_cells; }
+  unsigned get_num_nodes() const { return num_nodes; }
+  const Layout &get_cc_linkage() const { return cell_to_cell_linkage; }
+  const Boundary_Layout &get_cs_linkage() const { return cell_to_side_linkage; }
```

Comment:
This is a good question.  The intent is to have the layout in scope until it can be serialized and copied for the main solution step.  However, not all dependent projects might use it this way.